### ISSUE TITLE
Install with SSL false fails to generate valid 

### ISSUE THREAD

**Issue by dutchwhizzman** [othalla/puppet-nextcloud#4]
Oct 29 13:26:22 hostname.tld nginx[25829]: nginx: [emerg] "location" directive is not allowed here in /etc/nginx/sites-enabled/nextcloud_server_main.conf:25

Basically, the server {} definition is closed before the location / {} is put in the file.

**Comment by othalla** [comment]
Thank you for reporting it @dutchwhizzman 
i'll have a look!

**Comment by dutchwhizzman** [comment]
I have made some changes and improvements to get it working for me, want me to submit a push?

**Comment by othalla** [comment]
If you have the fix, you can make a PR yes !

**Comment by dutchwhizzman** [comment]
I found the root cause:

https://github.com/voxpupuli/puppet-nginx/issues/1142

You have to define ssl_only for the location part if you want it to be in
the https server, if it's not defined, it refuses to put it in the ssl part
of the server and puts it on top of the file. I thought I fixed it because
I only tested the non https version. Now I started re-integrating
https/certbot, I ran into the same thing again.

Now, how to solve it...? You can't do if {} inside a definition....

**Comment by othalla** [comment]
Hum i see, maybe put the whole block in a if?

**Comment by dutchwhizzman** [comment]
Yeah, I have a split for the ssl/non-ssl part now and the root location. I
still need to do the rest and test thoroughly before  will create an
updated pull request.
